Lib/ensurepip/__init__.py

```diff
@@ -26,9 +27,17 @@ def _run_pip(args, additional_paths=None):
     if additional_paths is not None:
         sys.path = additional_paths + sys.path
 
-    # Install the bundled software
-    import pip._internal
-    return pip._internal.main(args)
+    # Invoke pip as if it's the main module, and catch the exit.
+    backup_argv = sys.argv[:]
+    sys.argv[1:] = args
+    try:
+        runpy.run_module("pip", run_name="__main__", alter_sys=True)
```

Comment:
I had to check the effect of alter_sys=True. So I suggest to add a comment to explain the sys is left unmodified at run_module() exit.

```suggestion
        # run_module() alters sys.modules and sys.argv, but restore them at exit
        runpy.run_module("pip", run_name="__main__", alter_sys=True)
```